brave/brave-core
# BraveProfileLock

Resolves brave/brave-browser#423.

## Submitter Checklist:


## Test Plan:

Adapted from test plan in #101.

### Manual testing

1. You will need a Brave browser-laptop profile with some profile data saved in it.
2. Start Brave browser-laptop in the background.
3. `yarn start`
4. Open the main menu (e.g. **Brave** on macOS) and choose **Import Bookmarks and Settings...**
5. Choose your test Brave profile from the dropdown menu of browser profiles and click **Import**.
6. You should see an alert with the title "Close Brave," the text "To finish importing, close all Brave windows," and two buttons: **Try again** and **Cancel**.
7. **Cancel** should return you to the Settings page without importing anything from the Brave profile.
8. **Try again** should return you to the same alert repeatedly, until the background Brave process is quit.
9. Quit the background Brave process, then click **Try again**.
10. Your Brave profile data should be successfully imported.

### Cross-platform testing

I have verified that this PR builds and passes the manual tests described above on macOS. I am waiting on builds for Windows and Linux, and will update this issue with the results of those builds and manual tests once they're available.

## Reviewer Checklist:
